src/main/java/dev/jbang/cli/BaseScriptCommand.java

```diff
@@ -303,13 +309,20 @@ private static ScriptResource fetchFromURL(String scriptURL) {
 					throw new ExitException(10, exmsg);
 			}
 
-			scriptURL = swizzleURL(scriptURL);
-
-			String urlHash = Util.getStableID(scriptURL);
-			File urlCache = Settings.getCacheDir(Settings.CacheClass.urls).resolve(urlHash).toFile();
-			Path path = Util.downloadFileSwizzled(scriptURL, urlCache);
-
-			return new ScriptResource(scriptURL, urlCache, path.toFile());
+			if (Util.isGistURL(scriptURL)) {
+				GistCollection gistCollection = Util.extractFileFromGistWithSources(scriptURL);
+				File urlCache = Util.getUrlCache(gistCollection.getMainURL()).toFile();
+				Path path = Util.getPathToFile(gistCollection.getMainURL(), urlCache);
+				ScriptResource scriptResource = new ScriptResource(gistCollection.getMainURL(), urlCache,
+						path.toFile());
+				scriptResource.setResolvedSourcePaths(gistCollection.getResolvedSourcePathsAsList());
```

Comment:
okey so this whole block works but not a fan of `//SOURCES` now have two different behaviours on a set a file between if you have it in a gist vs having it on the filesystem.

Do you want to give it a try to extend this to be more generic or we change this to only work based on the central file driving `//SOURCES` ?

Reply:
It's because the focus of this PR was to solve scripts in Gists.
I didn't want to change other places as it would make the PR bigger and harder to review.

I was thinking about making `//SOURCES` work for multiple files for the the cases, but it's probably better in another PR.
We could then even apply convention over configuration:
  - If imported classes are in the expected path, then maybe `//SOURCES` would not even be needed.

Then I would not need `//SOURCES` here:
https://github.com/tivrfoa/flatbuffers-java-rust/blob/main/java/SampleBinary.java

```java
//SOURCES MyGame/Sample/Color.java
//SOURCES MyGame/Sample/Equipment.java
//SOURCES MyGame/Sample/Monster.java
//SOURCES MyGame/Sample/Vec3.java
//SOURCES MyGame/Sample/Weapon.java
```

Reply:
Wdym expected ? If there is no //SOURCE what would you use to know which files to look for ?

Reply:
Btw. I'm fine with making this pr just about gist multi file support. 

I just then would like to see the behaviour be at least consistent.

Reply:
> Wdym expected ? If there is no //SOURCE what would you use to know which files to look for ?

For example Color.java is in the *"expected"* path **relative** to the `main` class (`SampleBinary.java`):

https://github.com/tivrfoa/flatbuffers-java-rust/blob/main/java/MyGame/Sample/Color.java

Reply:
Yes so how would you know that file is there if not with a //SoURCES ref?

Reply:
Btw. I do want to add support for //SOURCES **/*.java but thats not configuration by convention which I dont See s way to do cleanly - curious to what approach you are considering ?

Reply:
> Yes so how would you know that file is there if not with a //SoURCES ref?

Beyond the current way of checking `//SOURCES`, I would check the imports if they exist in the "expected" path.

It may seem that would be to much work and inefficient, but I think it will be simple, because I would only check once for each prefix.

So in my example:
```java
import MyGame.Sample.Weapon;
import java.nio.ByteBuffer;
import java.io.FileOutputStream;
import java.nio.channels.FileChannel;
```
I would check MyGame, then MyGame.Sample, then MyGame.Sample.Weapon.

Then `java` would be checked only once for `java.nio.ByteBuffer`, and would mark that the relative path does not exist.

Reply:
Okey, so based on imports. That just have quite a few edge cases. I would go for //SoURCEs **/*.java instead. Checking every import will be quite slow.